v03_pipeline/lib/tasks/write_metadata_for_run.py

```diff
@@ -47,7 +47,7 @@ def output(self) -> luigi.Target:
     def complete(self) -> bool:
         return GCSorLocalTarget(self.output().path).exists()
 
-    def requires(self) -> luigi.Task:
+    def requires(self) -> list[luigi.Task]:
```

Comment:
🙇

Reply:
ah I just realized we need it in addition to `ruff`

Reply:
yeah, I have some context on why we don't have it right now if you want it, but probably better on a ticket.  

tldr neither hail nor luigi play well with strict type-checking (airflow does though I think).